File: backend/app/schemas/world.py
from typing import Literal

from pydantic import BaseModel, Field


ActionId = Literal["move", "rest", "work", "eat", "social"]


class WorldInfo(BaseModel):
    id: str
    name: str
    day: int = Field(ge=1)
    time: str
    tick: int = Field(ge=0)


class LocationInfo(BaseModel):
    id: str
    name: str
    description: str


class NpcStatus(BaseModel):
    energy: int = Field(ge=0, le=100)
    mood: int = Field(ge=0, le=100)
    social: int = Field(ge=0, le=100)


class NpcInfo(BaseModel):
    id: str
    name: str
    role: str
    personality: list[str]
    location_id: str
    current_action: ActionId
    status: NpcStatus


class WorldData(BaseModel):
    world: WorldInfo
    locations: list[LocationInfo]
    npcs: list[NpcInfo]

File: frontend/src/types/world.ts
export interface ApiResponse<T> {
  success: boolean
  data: T
  message: string
}

export interface WorldInfo {
  id: string
  name: string
  day: number
  time: string
  world_version: number
  clock_tick: number
  event_sequence: number
}

export interface LocationInfo {
  id: string
  name: string
  description: string
}

export interface NpcStatus {
  energy: number
  mood: number
  social: number
}

export type ActionId = 'move' | 'rest' | 'work' | 'eat' | 'talk' | 'wait'

export interface NpcInfo {
  id: string
  name: string
  role: string
  personality: string[]
  location_id: string
  current_action: ActionId
  status: NpcStatus
}

export interface WorldData {
  world: WorldInfo
  locations: LocationInfo[]
  npcs: NpcInfo[]
}
